src/app/clusters/camera-av-settings-user-level-management-server/camera-av-settings-user-level-management-server.cpp

```diff
@@ -0,0 +1,845 @@
+    }
+
+    if (SupportsOptAttr(OptionalAttributes::kTiltMin))
+    {
+        VerifyOrReturnError(HasFeature(Feature::kMechanicalTilt), CHIP_ERROR_INVALID_ARGUMENT,
+                            ChipLogError(Zcl,
+                                         "CameraAVSettingsUserLevelManagement[ep=%d]: Feature configuration error. If TiltMin is "
+                                         "enabled, then Mechanical Tilt feature is required",
+                                         mEndpointId));
+    }
+
+    if (SupportsOptAttr(OptionalAttributes::kTiltMax))
+    {
+        VerifyOrReturnError(HasFeature(Feature::kMechanicalTilt), CHIP_ERROR_INVALID_ARGUMENT,
+                            ChipLogError(Zcl,
+                                         "CameraAVSettingsUserLevelManagement[ep=%d]: Feature configuration error. If TiltMax is "
+                                         "enabled, then Mechanical Tilt feature is required",
+                                         mEndpointId));
+    }
+
+    if (SupportsOptAttr(OptionalAttributes::kPanMin))
+    {
+        VerifyOrReturnError(HasFeature(Feature::kMechanicalPan), CHIP_ERROR_INVALID_ARGUMENT,
+                            ChipLogError(Zcl,
+                                         "CameraAVSettingsUserLevelManagement[ep=%d]: Feature configuration error. If PanMin is "
+                                         "enabled, then Mechanical Pan feature is required",
+                                         mEndpointId));
+    }
+
+    if (SupportsOptAttr(OptionalAttributes::kPanMax))
+    {
+        VerifyOrReturnError(HasFeature(Feature::kMechanicalPan), CHIP_ERROR_INVALID_ARGUMENT,
+                            ChipLogError(Zcl,
+                                         "CameraAVSettingsUserLevelManagement[ep=%d]: Feature configuration error. If PanMax is "
+                                         "enabled, then Mechanical Pan feature is required",
+                                         mEndpointId));
+    }
+
+    LoadPersistentAttributes();
+
+    // Set default MPTZ
+    mMptzPosition.pan  = defaultPan;
+    mMptzPosition.tilt = defaultTilt;
+    mMptzPosition.zoom = defaultZoom;
+
+    VerifyOrReturnError(AttributeAccessInterfaceRegistry::Instance().Register(this), CHIP_ERROR_INTERNAL);
+    ReturnErrorOnFailure(CommandHandlerInterfaceRegistry::Instance().RegisterCommandHandler(this));
+    return CHIP_NO_ERROR;
+}
+
+bool CameraAvSettingsUserLevelMgmtServer::HasFeature(Feature feature) const
+{
+    return mFeature.Has(feature);
+}
+
+bool CameraAvSettingsUserLevelMgmtServer::SupportsOptAttr(OptionalAttributes aOptionalAttrs) const
+{
+    return mOptionalAttrs.Has(aOptionalAttrs);
+}
+
+// Mutators that may be invoked by a delegate in responding to command callbacks, or due to local on device changes
+//
+void CameraAvSettingsUserLevelMgmtServer::setPan(Optional<int16_t> pan)
+{
+    if (pan.HasValue())
+    {
+        mMptzPosition.pan = pan;
```

Comment:
Needs to mark dirty, no?

Also, why is setting an empty optional ignored?

Similar questions for other setters.